### PR TITLE
Cover routes-for-location spec test gaps

### LINKED ISSUE #1230: routes-for-location: Complete test coverage for edge cases
### Description

**Current behavior:**
The test suite for the `routes-for-location` endpoint is missing coverage for several edge cases and scenarios defined in the wiki spec (e.g., bounding box precedence when `latSpan`/`lonSpan` are provided, fallback to 500m vs 10km based on the presence of a query, and verifying random shuffling when `maxCount` is exceeded).

**Expected behavior (per spec):**
> Test coverage: main scenarios and edge cases from the spec are covered by tests

**What to change:**
1. Review `internal/restapi/routes_for_location_handler_test.go` and add missing test cases for all spec edge cases.
2. Specifically ensure coverage for `latSpan`/`lonSpan` overrides, default radius logic, random truncation, and the text search improvements.

### PR DESCRIPTION
**Pull request by ARCoder181105**
Closes #1230

Adds the spec cases the `routes-for-location` suite was missing. Tests only.

- `references.situations`: the alert-collection branch had no coverage — a route-scoped alert now
  has to surface as a situation reference.
- Default search radius: previously only inferred from the span tests, now pinned directly, so a
  change to the no-query default fails a test.
- `maxCount=0`: asserts the `fieldErrors` message the spec quotes, not just the 400.

Remaining spec cases (span precedence, query radius, randomized truncation, text search,
candidate cap, clamping, out-of-range, missing lat/lon) were already covered.

Stacked on #1357 and #1358 — this branch is built on top of both, so the diff will shrink to
just this change once they merge.

## Summary by CodeRabbit

- **Tests**
  - Expanded coverage for routes-for-location responses, including alert references and route-scoped situations.
  - Verified bounding-box sizing and the default 600-meter search radius.
  - Confirmed the default maximum result count of 10.
  - Added validation coverage for zero, negative, and invalid maximum-count values, including exact error messages.

### CODE REVIEW

**Review by Ahmedhossamdev — COMMENTED** [review]
Test coverage is solid and the behavior changes match Java (default maxCount
10 with the 50 clamp, includeReferences gating, and the exact maxCount error
message all check out against RoutesForLocationAction.java and against the
hosted Java API). Gate is green.

One thing to fix before merge: the default-radius test pins 600m, but Java's
no-query default is 500m and the hosted Java returns exactly the radius=500
set for a no-parameter request. Change DefaultSearchRadiusInMeters to 500 and
the test then pins the right value. Note the diff is not test-only: it also
carries the maxCount default and includeReferences behavior changes.

**Comment by aaronbrethorst** [comment]
### Code review

Found 1 issue:

1. `TestRoutesForLocationDefaultRadiusMatches600Meters` pins the no-query default radius at 600m and documents that as the spec value, but the spec this PR is closing to (#1230) says the fallback is "500m vs 10km based on the presence of a query" — and the hosted Java API returns `radius=500` for a no-parameter request. `models.DefaultSearchRadiusInMeters = 600` is the current (wrong) value, so this test locks in the divergence and will have to be edited again when the constant is corrected. The 10km query fallback (`QuerySearchRadiusInMeters`) already matches the spec, which makes 600 look like a typo rather than an intentional choice. Either fix the constant to 500 in this PR and pin that, or rename the test/comment so it reads as "current behavior" instead of "the spec default".

https://github.com/OneBusAway/maglev/blob/1824d6d03f061afb14dc9750fde31cc3dc962c31/internal/restapi/routes_for_location_handler_test.go#L181-L208

🤖 Generated with [Claude Code](https://claude.ai/code)

<sub>- If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by aaronbrethorst — DISMISSED** [review]
The tests here are genuinely good, which is why I want to get one detail right
before it lands.

What works well: `TestRoutesForLocationHandlerSituationReferences` seeds a real
route-scoped alert and asserts on its actual ID, and then re-requests with
`includeReferences=false` to prove suppression against a populated source
rather than an empty one — that second half is the part most people skip, and
it's what makes the assertion mean something. And pinning the radius by
*equivalence* (`default == radius=600`, `default ⊃ radius=500`) instead of
hardcoding a distance is a nice technique; it fails for the right reason.

**The blocker — `TestRoutesForLocationDefaultRadiusMatches600Meters`
(internal/restapi/routes_for_location_handler_test.go:182-208):**

The test and its doc comment pin 600m as "the no-query default radius," but
600 looks like a bug, not the spec. Issue #1230 — the issue this PR is closing —
says the fallback is "500m vs 10km based on the presence of a query," and
@Ahmedhossamdev verified the hosted Java API returns exactly the `radius=500`
set for a no-parameter request. `QuerySearchRadiusInMeters = 10000` already
matches the 10km half, which makes `DefaultSearchRadiusInMeters = 600` the odd
one out. Locking 600 in with a test named `Matches600Meters` makes it *harder*
to fix later, because the fix now also has to argue with a test that says the
current value is correct.

One correction to the suggestion in the thread, though: please **don't** just
change the constant to 500 in this PR.
`models.DefaultSearchRadiusInMeters` is shared — `internal/gtfs/location_params.go:30`
uses it as the `BoundsFromParams` fallback, so flipping it silently changes the
default search area for stops-for-location and trips-for-location too. That's a
real behavior change across three endpoints and deserves its own PR and its own
tests, not a drive-by in a test-coverage change.

So what I'd like here is the narrow version: drop the 600m test from this PR
(or keep it and reword the name and comment to say it pins *current* behavior,
not the spec, with a `TODO` referencing a new issue), and open a separate issue
for the 500-vs-600 discrepancy noting the shared-constant blast radius. I'm
happy to take that follow-up PR.

Two smaller notes, neither blocking:

- `TestRoutesForLocationHandlerMaxCountZeroFieldErrorMessage` sits right next to
  the existing `TestRoutesForLocationHandlerMaxCountLessThanOrEqualZero`, which
  already covers `maxCount=0`. CONTRIBUTING.md prefers table-driven tests for
  multiple cases of the same handler — folding the message assertion into the
  existing test would read better.
- Its comment says "the spec quotes" that message, but `testdata/openapi.yml`
  has no `fieldErrors` schema; the string only exists in
  `internal/utils/api.go`. Worth softening the wording.

Also, ordering: this is stacked on #1357 and #1358. I've approved both, so once
they land, rebase and this is a short round trip.

**Comment by ARCoder181105** [comment]
On opening a separate issue for the default radius: #1227 already tracks this. I closed it on 2026-08-01 accepting the global constant, and @burma-shave noted the deviation would instead be recorded in the spec's Implementation Decisions section. That entry never landed — the live wiki page still states 500m, and its Implementation Decisions section only covers `query` matching.

So it is either reopening #1227 or filing a new issue that supersedes it. Happy to do whichever you prefer.

Blast radius for whichever lands: `models.DefaultSearchRadiusInMeters` is the `BoundsFromParams` fallback (`internal/gtfs/location_params.go:30`), so changing it to 500 also moves the default search area for stops-for-location and trips-for-location, along with their fixture tests.

File: internal/restapi/routes_for_location_handler_test.go [hunk 1]
```diff
@@ -67,6 +68,43 @@ func TestRoutesForLocationHandlerIncludeReferences(t *testing.T) {
...
 
+// TestRoutesForLocationHandlerSituationReferences verifies that a route-scoped
+// alert surfaces as a situation reference — the alert-collection branch feeding
+// data.references.situations had no coverage before this test.
+//
+// This is current maglev behavior, not the documented spec: the routes-for-location
+// wiki page states that references.situations is present but empty for this endpoint,
+// and its Implementation Decisions section has no entry recording a deviation. The
+// population comes from commit 4e09c1c, which added it across all non-trip handlers
+// (route, route-search, stops-search) rather than specifically for this endpoint.
```
[the remark this answers is not in the document]
Comment (ARCoder181105) [reply internal/restapi/routes_for_location_handler_test.go:79]: I think this is stale

**Comment by ARCoder181105** [comment]
Pushed two commits addressing this.

`TestRoutesForLocationDefaultRadiusMatches600Meters` → renamed to
`TestRoutesForLocationDefaultRadiusMatchesCurrentConstant`, comment now says
explicitly this pins current behavior (models.DefaultSearchRadiusInMeters),
not the spec, with the 500m wiki/hosted-Java value noted and a link to #1227.
Assertions unchanged — verified locally that flipping the constant to 500
makes the test fail, so it's not vacuous. (1c6edbe)

Folded the three maxCount tests into one table-driven test, each case
carrying the expected fieldErrors message. (2672197)

On "the spec quotes" wording — one correction: the wiki page does quote
"must be greater than zero" in extension 1a, so the comment wasn't wrong,
just ambiguous about which spec. Reworded to name the wiki page directly;
testdata/openapi.yml is the one with no fieldErrors schema.

Also flagged, separately: TestRoutesForLocationHandlerSituationReferences
pins references.situations being populated here, but the wiki says that
array is empty for this endpoint and has no Implementation Decisions entry
for it. Comment now notes it's current behavior from 4e09c1c (applied across
all non-trip handlers), not something the spec documents. Left the
assertions as-is per the same reasoning as the radius test.

On the issue: commented separately that #1227 already tracks the 600-vs-500
gap, closed 2026-08-01 as an accepted deviation with a note that it'd be
recorded in Implementation Decisions — never happened. Asked there whether
to reopen or file a new one.

#1357 and #1358 are merged, so this is genuinely tests-only now.
